comit/src/btsieve/bitcoin/cache.rs

```diff
@@ -69,3 +72,44 @@ where
         Ok(block)
     }
 }
+
+impl<C> Cache<C>
```

Comment:
This goes against the original design of the cache being a proxy.

This functionality doesn't need the cache so should be implemented just on the abstraction of a connector.
If you pass a `Cache<Connector`, you won't loose any performance while computing this assuming the blocks are already cached.

Reply:
> This goes against the original design of the cache being a proxy.

I don't understand this.

> This functionality doesn't need the cache so should be implemented just on the abstraction of a connector.

I don't understand this either, we want the cache so that if this function is called repeatedly (e.g. by a user polling the REST API) the blocks are cached.

> If you pass a `Cache<Connector`, you won't loose any performance while computing this assuming the blocks are already cached.

I don't get what you are suggesting here? Pass the `Cache<Connetor>` instead of implement this as a method on the `Cache` is what I understand but that makes no sense to me?

Reply:
>> This goes against the original design of the cache being a proxy.

> I don't understand this.

A proxy, by design, forwards calls to something else and adds additional logic around it while adhering to the same interface that than the thing it forwards to.
This function does not forward to the connector but instead composes new functionality from the connector. A proxy must adhere to the same interface as the thing it is forwarding to for it to be a proxy.

> I don't understand this either, we want the cache so that if this function is called repeatedly (e.g. by a user polling the REST API) the blocks are cached.

This function can just have the signature: `median_time(connector: C)`. No trait, no impl block, just a function with a generic parameter.

If the caller of this fn wants it to be fast, they should pass a `Cache<Connector` instead of a plain connector but there is no need for this function to know about caching. It only depends on the interface of a connector.

If we stick to the cache being a proxy, the function doesn't need to care about which particular connector implementation you are passing and whether or not it is wrapped in a cache.